retrieval_service/datastore/providers/firestore.py

```diff
@@ -35,47 +38,111 @@ class Config(BaseModel, datastore.AbstractConfig):
 
 class Client(datastore.Client[Config]):
     __client: AsyncClient
+    policies_collection: AsyncQuery
+    amenities_collection: AsyncQuery
 
     @datastore.classproperty
     def kind(cls):
         return FIRESTORE_IDENTIFIER
 
     def __init__(self, client: AsyncClient):
         self.__client = client
+        self.policies_collection = AsyncQuery(self.__client.collection("policies"))
+        self.amenities_collection = AsyncQuery(self.__client.collection("amenities"))
```

Comment:
these should probably be reduced visibility -- either "__x_collection" or "_x_collection".